### PR TITLE
Level 1 measurement support in Sampler options

### PR DESCRIPTION
**Pull request by gadial**
### Summary
This PR adds support for level 1 measurement options in `SamplerV2`

### Details and comments
This PR extends `ExecutionOptionsV2` to a new class, `SamplerExecutionOptionsV2` having an additional field `meas_type` which can accept the values `"classified", "kerneled", "avg_kerneled"`, where "classified" represents the usual measurement level 2 output, and the other options represent measurement level 1 output.

Fixes part of #1554

### CODE REVIEW

File: qiskit_ibm_runtime/options/sampler_execution_options.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    * ``"kerneled"``: A complex NumPy array of shape ``(*pub_shape, num_shots, creg_size)``, where
+      each entry represents an IQ data point (resulting from kerneling the measurement trace) in 
+      arbitrary units.
+
+    * ``"avg_kerneled"``: A complex NumPy array of shape ``(*pub_shape, creg_size)``, where
+      each entry represents an IQ data point (resulting from kerneling the measurement trace and 
+      averaging over shots) in arbitrary units. This option is equivalent to selecting
+      ``"kerneled"`` and then averaging over the shots axis, but requires less data bandwidth.
+
+    See https://pubs.aip.org/aip/rsi/article/88/10/104703/836456 for a description of kerneling.
```
Comment (ihincks) [inline qiskit_ibm_runtime/options/sampler_execution_options.py:43]: please turn this into an ReST link https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages

**Review by ihincks — COMMENTED** [review]
Thanks @gadial , one minore comment.

More generally, is the testing here in line with the standards of this library? I have an `IQBackendWrapper < Backend` class implement elsewhere if we want to use it.

**Comment by gadial** [comment]
> More generally, is the testing here in line with the standards of this library? I have an `IQBackendWrapper < Backend` class implement elsewhere if we want to use it.

I added relevant tests for this PR to the existing ones; there are no extensive tests as far as I can see. However, Such a wrapper sounds like a good idea and I can add more exhaustive tests around it.

File: qiskit_ibm_runtime/options/sampler_execution_options.py (hunk 1 above)
Comment (gadial) [reply qiskit_ibm_runtime/options/sampler_execution_options.py]: Done

**Review by ihincks — APPROVED** [review]

**Pull request merged**